modules/imgcodecs/src/loadsave.cpp

```diff
@@ -633,12 +638,20 @@ Mat imread( const String& filename, int flags )
     Mat img;
 
     /// load the data
-    imread_( filename, flags, img );
+    imread_( filename, flags, img, true);
 
     /// return a reference to the data
     return img;
 }
 
+void imread( const String& filename, OutputArray & dst, int flags )
+{
+    CV_TRACE_FUNCTION();
+
+    /// load the data
+    imread_(filename, flags, dst.getMatRef(), false);
```

Comment:
> `.getMatRef`

It should not be used in general because it restricts parameter type to `Mat` only.